src/System.Private.CoreLib/src/System/HighPerformanceCounter.Unix.cs

```diff
@@ -2,13 +2,32 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+
 namespace System
 {
     internal static class HighPerformanceCounter
     {
-        public static ulong TickCount => Interop.Sys.GetHighPrecisionCount();
+        public static ulong TickCount
+        {
+            get
+            {
+                long tickCount;
+                bool success = Interop.Sys.GetTimestamp(out tickCount);
+                Debug.Assert(success);
+                return (ulong)tickCount;
+            }
+        }
+
+        public static ulong Frequency { get; private set; }
 
-        // Cache the frequency on the managed side to avoid the cost of P/Invoke on every access to Frequency
-        public static ulong Frequency { get; } = Interop.Sys.GetHighPrecisionCounterFrequency();
+        static HighPerformanceCounter()
```

Comment:
Explicit static constructors are not good for performance.

Reply:
What is the recommended pattern then? LazyInitializer? This class has just two fields and anyone using it is bound to use both, so I assumed it would be ok.

Reply:
Non-explicit before field constructor like what was there before or like what is in the Windows version.

Reply:
The Windows version has the same `out` parameter: https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/HighPerformanceCounter.Windows.cs#L22